pkg/xray/resource_xray_watch.go

```diff
@@ -0,0 +1,118 @@
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "Nested argument describing the resources to be watched. Defined below.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:             schema.TypeString,
+							Required:         true,
+							Description:      "Type of resource to be watched. Options: `all-repos`, `repository`, `build`, `project`, `all-projects`.",
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"all-repos", "repository", "build", "project", "all-projects"}, true)),
+						},
+						"bin_mgr_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Default:     "default",
+							Description: "The ID number of a binary manager resource. Should be set to `default` if not set on the Artifactory side.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The name of the build or repository. Enable Xray indexing must be enabled on the repo or build",
+						},
+						"filter": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "Nested argument describing filters to be applied. Defined below.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:             schema.TypeString,
+										Required:         true,
+										Description:      "The type of filter, such as `regex`, `package-type` or `ant-patterns`",
+										ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"regex", "package-type", "ant-patterns"}, true)),
+									},
+									// TODO support Exclude and Include patterns
+									// eg "value":{"ExcludePatterns":[],"IncludePatterns":["*"]}
+									"value": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "The value of the filter, such as the text of the regex or name of the package type.",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			// "assigned_policies" in the API call body. Plural is replaced for better reflection of the
+			// actual functionality (see HCL examples)
+			"assigned_policy": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "Nested argument describing policies that will be applied. Defined below.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The name of the policy that will be applied",
+						},
+						"type": {
+							Type:             schema.TypeString,
+							Required:         true,
+							Description:      "The type of the policy",
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"security", "license"}, true)),
+						},
+					},
+				},
+			},
+			"watch_recipients": {
+				Type:        schema.TypeList,
```

Comment:
a list would allow us (potentially?) to send the same notification twice. I mean, maybe RT deal with this but a List allows duplicates and I can't see why

Reply:
yeah, the problem is the RT doesn't deal with it. You can easily use the same email several times in the API call. 
Example of the watch created with REST API: 
![image](https://user-images.githubusercontent.com/55418063/144948189-d51079b9-e3ef-49bd-a37b-2e12e4dbd716.png)

At the same time, UI doesn't allow users to add duplicates. I've tried to add dup to the existing watch:  
![image](https://user-images.githubusercontent.com/55418063/144948314-546f6580-3570-43b0-8310-e0a8583f20de.png)
Also, I completely forgot to add that `watch_recipients` to the resource!

Reply:
Then definitely change it from a list to set - it looks like the UI is doing client-side validation (which is basically the clearest sign yet that you need to also do client side validation in TF)

Reply:
it is set now